core/app/models/spree/variant.rb

```diff
@@ -24,7 +24,6 @@ class Variant < Spree::Base
       stock_items.discard_all
       images.destroy_all
       prices.discard_all
-      currently_valid_prices.discard_all
```

Comment:
This is not needed because the line above would still remove those, right?

Reply:
Right!